src/util/simplify_expr_boolean.cpp

```diff
@@ -245,7 +245,9 @@ bool simplify_exprt::simplify_not(exprt &expr)
   }
   else if(op.id()==ID_exists) // !(exists: a) <-> forall: not a
   {
-    assert(op.operands().size()==2);
+    DATA_INVARIANT(
+      op.operands().size() == 2,
+      "exists expression has two parameters");
```

Comment:
Please introduce `to_exists_expr` which should do this sort of check and use it here.